test/integration/e2e_test.go

```diff
@@ -270,4 +275,26 @@ func TestE2EFullIntegration(t *testing.T) {
 	require.NoError(t, err)
 	require.Greaterf(t, len(bytes.Split(kubectlOutput2, []byte("\n"))), 2, "expected some namespaces to be returned again")
 	t.Logf("second kubectl command took %s", time.Since(start).String())
+
+	// probe our cache for the current ID token as a proxy for a whoami API
+	cache := filesession.New(sessionCachePath, filesession.WithErrorReporter(func(err error) {
+		require.NoError(t, err)
+	}))
+
+	downstreamScopes := []string{coreosoidc.ScopeOfflineAccess, coreosoidc.ScopeOpenID, "pinniped:request-audience"}
+	sort.Strings(downstreamScopes)
+	token := cache.GetToken(oidcclient.SessionCacheKey{
+		Issuer:      downstream.Spec.Issuer,
+		ClientID:    "pinniped-cli",
+		Scopes:      downstreamScopes,
+		RedirectURI: "http://localhost:0/callback",
+	})
+	require.NotNil(t, token)
+
+	idTokenClaims := token.IDToken.Claims
+	username := idTokenClaims[oidc.DownstreamUsernameClaim].(string)
+	groups, _ := idTokenClaims[oidc.DownstreamGroupsClaim].([]string)
+
+	require.Equal(t, env.SupervisorTestUpstream.Username, username)
+	require.Equal(t, env.SupervisorTestUpstream.ExpectedGroups, groups) // TODO wire ExpectedGroups to env var config
```

Comment:
@ankeesler This should be the last outstanding item for the Okta bits.

Reply:
ACK.